model/buildlogger_storage.go

```diff
@@ -27,18 +27,35 @@ func (lf LogFormat) Validate() error {
 	}
 }
 
-// LogArtifact is a type that describes a sub-bucket of logs stored in s3. It
-// is the bridge between S3-based offline log storage and the cedar-based log
-// metadata storage. The prefix field indicates the name of the sub-bucket. The
-// top level bucket is accesible via the cedar.Environment interface.
+// LogArtifact describes a sub-bucket of logs stored in s3. It is the bridge
+// between S3-based offline log storage and the cedar-based log metadata
+// storage. The prefix field indicates the name of the sub-bucket. The top
+// level bucket is accesible via the cedar.Environment interface.
 type LogArtifactInfo struct {
 	Prefix      string             `bson:"prefix"`
 	Permissions pail.S3Permissions `bson:"permissions"`
 	Version     int                `bson:"version"`
+	Chunks      []LogChunkInfo     `bson:"chunks"`
 }
 
 var (
 	logArtifactInfoPrefixKey      = bsonutil.MustHaveTag(LogArtifactInfo{}, "Prefix")
 	logArtifactInfoPermissionsKey = bsonutil.MustHaveTag(LogArtifactInfo{}, "Permissions")
 	logArtifactInfoVersionKey     = bsonutil.MustHaveTag(LogArtifactInfo{}, "Version")
+	logArtifactInfoChunksKey      = bsonutil.MustHaveTag(LogArtifactInfo{}, "Chunks")
+)
+
+// LogChunkInfo describes a chunk of log lines stored in s3.
```

Comment:
in a bucket? could use a different backend?

Reply:
This artifact is coupled pretty tightly with S3

Reply:
can we relax it a bit?

Reply:
Theoretically, given how the application works, (i.e. the protections for content in s3 is all read by the application and then returned to the users from the application, so our security model is really external to the bucket, which I think would  resolve the concerns here. 

Put another way, access to the s3 bucket without the application is not likely to be particularly useful.